src/lib/core/popen.c

```diff
@@ -840,6 +892,9 @@ popen_delete(struct popen_handle *handle)
 		popen_unregister(handle);
 	}
 
+	/* Unblock fibers waiting in popen_wait_timeout(). */
+	fiber_cond_broadcast(&handle->completion_cond);
```

Comment:
This works with `ph:wait()`, but not with `ph:read` or `ph:write`, see https://github.com/tarantool/tarantool/issues/7653#issuecomment-1239094646. I think we need to come up with a better way to handle use-after-free.

Reply:
Answered in https://github.com/tarantool/tarantool/issues/7653#issuecomment-1239658685. I think that we can solve it together with #7654. I don't think that we're obligated to do everything in one patchset, however we can.

Reply:
Fair enough.